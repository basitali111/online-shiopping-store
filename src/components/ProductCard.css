.card-container {
  border: 1px solid black;
  border-radius: 18px;
  list-style: none;
  margin-top: 40px;
  width: 100%;
  height: 400px;
  padding: 0;
}

.text-info {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin-top: 30px;
  margin-left: 3px;
  gap: 20px;
  padding-bottom: 30px;
}

.card-info {
  padding-top: 10px;
}
