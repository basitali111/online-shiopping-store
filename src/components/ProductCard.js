/* eslint-disable */
import '../components/ProductCard.css'
const ProductCard = (props) => {
  const {
    id, title, price, image,
  } = props;

  return (
 
   <li className='card-container'>
    <div className='card-info'>
      <div className="image-product">
        <img src={image} alt={title} width="220" height="200"/>
      </div>
      <div className='text-info'>
        <p>{title}</p>
        <p>
          $
          {price}
        </p>
      </div>
      </div>
   </li>
  );
};

export default ProductCard;
