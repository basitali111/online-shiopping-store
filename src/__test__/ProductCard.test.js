import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ProductCard from '../components/ProductCard';

test('should match the product card props to Snapshot', () => {
  const data = [{
    id: 1,
    title: 'basit',
    image: 'image',
    price: '45',
  }];
  const tree = render(
    <BrowserRouter>
      <ProductCard data={data} />
    </BrowserRouter>,
  );
  expect(tree).toMatchSnapshot();
});
