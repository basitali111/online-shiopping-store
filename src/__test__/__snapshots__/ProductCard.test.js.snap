// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the home page Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="card-container"
      >
        <div
          class="image-product"
        >
          <img
            height="220"
            width="220"
          />
        </div>
        <div>
          <p />
          <p>
            $
          </p>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="card-container"
    >
      <div
        class="image-product"
      >
        <img
          height="220"
          width="220"
        />
      </div>
      <div>
        <p />
        <p>
          $
        </p>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should match the product card props to Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="card-container"
      >
        <div
          class="image-product"
        >
          <img
            height="220"
            width="220"
          />
        </div>
        <div
          class="text-info"
        >
          <p />
          <p>
            $
          </p>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="card-container"
    >
      <div
        class="image-product"
      >
        <img
          height="220"
          width="220"
        />
      </div>
      <div
        class="text-info"
      >
        <p />
        <p>
          $
        </p>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
